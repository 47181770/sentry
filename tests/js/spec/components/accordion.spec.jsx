import React from 'react';
import {mount, shallow} from 'enzyme';
import Accordion from 'app/components/accordion';

describe('Accordion', function() {
  let routerContext = TestStubs.routerContext();

  it('renders', function() {
    let component = shallow(
      <Accordion priority="primary" size="large" labelClosed="Show More" cutoff={2}>
        <div key="1" />
        <div key="2" />
        <div key="3" />
      </Accordion>
    );
    expect(component).toMatchSnapshot();
  });

  it('has a working cutoff', function() {
    let component = shallow(
      <Accordion priority="primary" size="large" labelClosed="Show More" cutoff={1}>
        <div data-test-id="accordion-count" key="1" />
        <div data-test-id="accordion-count" key="2" />
        <div data-test-id="accordion-count" key="3" />
      </Accordion>
    );
    expect(component.find('[data-test-id="accordion-count"]')).toHaveLength(1);
  });

  it('opens', function() {
    let component = mount(
      <Accordion priority="primary" size="large" labelClosed="Show More" cutoff={1}>
        <div data-test-id="accordion-count" key="1" />
        <div data-test-id="accordion-count" key="2" />
        <div data-test-id="accordion-count" key="3" />
      </Accordion>,
      routerContext
    );
    component.find('Button').simulate('click');
    expect(component.find('[data-test-id="accordion-count"]')).toHaveLength(3);
  });

  it('closes', function() {
    let component = mount(
      <Accordion priority="primary" size="large" labelClosed="Show More" cutoff={1}>
        <div data-test-id="accordion-count" key="1" />
        <div data-test-id="accordion-count" key="2" />
        <div data-test-id="accordion-count" key="3" />
      </Accordion>,
      routerContext
    );
    component
      .find('Button')
      .simulate('click')
      .simulate('click');
    expect(component.find('[data-test-id="accordion-count"]')).toHaveLength(1);
  });
});
