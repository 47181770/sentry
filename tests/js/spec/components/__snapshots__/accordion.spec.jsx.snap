// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders 1`] = `
Array [
  <div
    key=".$1"
  />,
  <div
    key=".$2"
  />,
  <div
    key=".$3"
  />,
]
`;
